weaver/utils.py

```diff
@@ -346,33 +368,90 @@ def repr_json(data, force_string=True, **kwargs):
 
 
 def wait_secs(run_step=-1):
+    # type: (int) -> int
+    """
+    Obtain a wait time in seconds within increasing delta intervals based on iteration index.
+    """
     secs_list = (2, 2, 2, 2, 2, 5, 5, 5, 5, 5, 10, 10, 10, 10, 10, 20, 20, 20, 20, 20, 30)
     if run_step >= len(secs_list):
         run_step = -1
     return secs_list[run_step]
 
 
-def expires_at(hours=1):
-    # type: (Optional[int]) -> int
-    return now_secs() + hours * 3600
-
-
-def localize_datetime(dt, tz_name="UTC"):
+def localize_datetime(dt, tz_name=None):
     # type: (datetime, Optional[str]) -> datetime
     """
-    Provide a timezone-aware object for a given datetime and timezone name.
+    Provide a timezone-aware datetime for a given datetime and timezone name.
+
+    .. warning::
+        Any datetime provided as input that is not already timezone-aware will be assumed to be relative to the
+        current locale timezone. This is the default returned by naive :class:`datetime.datetime` instances.
+
+    If no timezone name is provided, the timezone-aware datatime will be localized with locale timezone offset.
+    Otherwise, the desired localization will be applied with the specified timezone offset.
     """
     tz_aware_dt = dt
     if dt.tzinfo is None:
-        utc = pytz.timezone("UTC")
-        aware = utc.localize(dt)
-        timezone = pytz.timezone(tz_name)
-        tz_aware_dt = aware.astimezone(timezone)
-    else:
+        tz_aware_dt = dt.astimezone()  # guess local timezone
+    if tz_name is None:
+        return tz_aware_dt
```

Comment:
Actually, it would depend if the input datetime already has the timezone information or not. 
If not (`None`), datetime package assumes it is local machine time. 
Otherwise, the original tz would remain in the returned datetime, but this is not really a problem because the important thing here is to apply tz info to make sure they can be localized properly as needed. 
There is not really any request parsing/input at the moment from the API to localize times to the user's  timezone, so it will not impact for the moment.